modelon/impact/client/client.py

```diff
@@ -149,19 +150,25 @@ def __init__(
         self._uri = URI(url)
         self._sal = modelon.impact.client.sal.service.Service(self._uri, context)
 
-        try:
-            self._validate_compatible_api_version()
-        except modelon.impact.client.sal.exceptions.AccessingJupyterHubError:
-            self._uri, jupyter_context = modelon.impact.client.jupyterhub.authorize(
-                self._uri,
-                interactive,
-                context,
-                jupyterhub_credential_manager,
-            )
+        if self._sal.is_jupyterhub_url():
+            if os.environ.get("JUPYTERHUB_SERVICE_PREFIX") is not None:
```

Comment:
This line is to check if we are inside the notebook or not I assume? It would be nice if this was clarified with a function explain this. So it reads:
```
if running in the jupyter notebook:
   stuff
```